src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -58,12 +58,38 @@ public byte[] mask(byte[] input) {
      */
     private void visitValue(MaskingState maskingState, @CheckForNull KeyMaskingConfig keyMaskingConfig) {
         skipWhitespaceCharacters(maskingState);
-        if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitArray(maskingState, keyMaskingConfig);
-        } else if (AsciiCharacter.isCurlyBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitObject(maskingState, keyMaskingConfig);
-        } else {
-            visitPrimitive(maskingState, keyMaskingConfig);
+        switch (maskingState.byteAtCurrentIndex()) {
```

Comment:
Might be nice to add a comment to each switch statement that it is deliberately a switch statement for improved performance

Reply:
Done